src/api/status/public/assets/js/gitHubStats.js

```diff
@@ -0,0 +1,80 @@
+    headers: {
+      Accept: 'application/vnd.github.v3+json',
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      weeklyCommits = data.all[data.all.length - 1];
+      document.getElementById(`weekly-commits-${repo}`).innerHTML = weeklyCommits;
+
+      document.getElementById(`yearly-commits-${repo}`).innerHTML = data.all.reduce(
+        (a, b) => a + b
+      );
+    })
+    .catch((error) => console.log(error));
+
+  // get weekly commits activity: https://docs.github.com/en/rest/reference/repos#get-the-weekly-commit-activity
+  fetch(`https://api.github.com/repos/Seneca-CDOT/${repo}/stats/code_frequency`, {
+    headers: {
+      Accept: 'application/vnd.github.v3+json',
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      if (weeklyCommits) {
+        document.getElementById(`lines-added-${repo}`).innerHTML = data[data.length - 1][1];
+        document.getElementById(`lines-removed-${repo}`).innerHTML = data[data.length - 1][2];
+      } else {
+        document.getElementById(`new-lines-${repo}`).innerHTML =
+          '<p class="mb-0">No commits this week<p/>';
+      }
+    })
+    .catch((error) => console.log(error));
+
+  // Get latest author from commits list: https://docs.github.com/en/rest/reference/repos#list-commits
+  fetch(`https://api.github.com/repos/Seneca-CDOT/${repo}/commits`, {
+    headers: {
+      Accept: 'application/vnd.github.v3+json',
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      const avatar = data[0].author.avatar_url;
+      const username = data[0].author.login;
+      const fullName = data[0].commit.author.name;
+
+      document.getElementById(`latest-author-image-${repo}`).src = avatar;
+      document.getElementById(`latest-author-image-${repo}`).alt = username;
+      document.getElementById(`latest-author-${repo}`).innerHTML = fullName;
+      document.getElementById(`latest-author-${repo}`).title = username;
+      document.getElementById(`latest-author-${repo}`).href = `https://github.com/${username}`;
+      document.getElementById(`latest-author-link-${repo}`).title = username;
+      document.getElementById(`latest-author-link-${repo}`).href = `https://github.com/${username}`;
+    })
+    .catch((error) => console.log(error));
+
+  // get total contributors
+  fetch(`https://api.github.com/repos/Seneca-CDOT/${repo}/contributors?per_page=1`, {
+    headers: {
+      Accept: 'application/vnd.github.v3+json',
+    },
+  })
+    .then((response) => response.headers.get('link'))
+    .then((data) => {
+      const contributors = data.match(/.*"next".*&page=([0-9]*).*"last".*/)[1];
+      document.getElementById(`total-contributors-${repo}`).innerHTML = contributors;
+    })
+    .catch((error) => console.log(error));
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  fetchGitHubData('telescope');
+  fetchGitHubData('satellite');
+});
+
+response.headers.get('link');
```

Comment:
What is this doing down here?  Where does `response` come from?

Reply:
oops not sure how that ended up there 😄